packages/nx/src/native/tui/app.rs

```diff
@@ -774,17 +939,442 @@ impl App {
+        self.recalculate_layout_areas();
+        // Ensure the pty instances get resized appropriately (no debounce as this is based on an imperative user action)
+        let _ = self.handle_pty_resize();
+    }
+
+    /// Forward key events to the currently focused pane, if any.
+    fn handle_key_event(&mut self, key: KeyEvent) -> io::Result<()> {
+        if let Focus::MultipleOutput(pane_idx) = self.focus {
+            let terminal_pane_data = &mut self.terminal_pane_data[pane_idx];
+            terminal_pane_data.handle_key_event(key)
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Returns true if the currently focused pane is in interactive mode.
+    fn is_interactive_mode(&self) -> bool {
+        match self.focus {
+            Focus::MultipleOutput(pane_idx) => self.terminal_pane_data[pane_idx].is_interactive(),
+            _ => false,
+        }
+    }
+
+    /// Ensures that the PTY instances get resized appropriately based on the latest layout areas.
+    fn debounce_pty_resize(&mut self) -> io::Result<()> {
+        // Get current time in milliseconds
+        let now = std::time::SystemTime::now()
+            .duration_since(std::time::UNIX_EPOCH)
+            .unwrap()
+            .as_millis();
+
+        // If we have a timer and it's not expired yet, just return
+        if let Some(timer) = self.resize_debounce_timer {
+            if now < timer {
+                return Ok(());
+            }
+        }
+
+        // Set a new timer for 200ms from now
+        self.resize_debounce_timer = Some(now + 200);
+
+        // Process the resize
+        self.handle_pty_resize()
+    }
+
+    /// Actually processes the resize event by updating PTY dimensions.
+    fn handle_pty_resize(&mut self) -> io::Result<()> {
+        if !self.layout_areas.is_some() {
+            return Ok(());
+        }
+
+        let mut needs_sort = false;
+
+        for (pane_idx, pane_area) in self
+            .layout_areas
+            .as_ref()
+            .unwrap()
+            .terminal_panes
+            .iter()
+            .enumerate()
         {
-            tasks_list.set_cloud_message(message);
+            let terminal_pane_data = &mut self.terminal_pane_data[pane_idx];
+            if let Some(pty) = terminal_pane_data.pty.as_ref() {
+                let (pty_height, pty_width) = TerminalPane::calculate_pty_dimensions(*pane_area);
+
+                // Get current dimensions before resize
+                let old_rows = if let Some(screen) = pty.get_screen() {
+                    let (rows, _) = screen.size();
+                    rows
+                } else {
+                    0
+                };
+                let mut pty_clone = pty.as_ref().clone();
+                pty_clone.resize(pty_height, pty_width)?;
+
+                // If dimensions changed, mark for sort
+                if old_rows != pty_height {
+                    needs_sort = true;
```

Comment:
future: This confuses me.. does the layout manager not know if it will change the size of a pty? When would the pty size change.. without the layout manager changing its size?